### PR TITLE
Week2 5 datastore

### PR DESCRIPTION
**Pull request by KhawYewOnn**
Complete week 2 final requirement

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -13,20 +13,72 @@
...
 package com.google.sps.servlets;
-
+import com.google.gson.Gson;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.*;
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: Avoid folded imports

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -13,20 +13,72 @@
...
+import java.util.ArrayList;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: Extra blank line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -13,20 +13,72 @@
...
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<String> messages;
+  
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:39]: Can be initialized with declaration.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -13,20 +13,72 @@
...
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    messages = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String message = (String) entity.getProperty("message");
+      messages.add(message);
+    }
+
+    response.setContentType("application/json");
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: content type can be constant

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -13,20 +13,72 @@
...
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    messages = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String message = (String) entity.getProperty("message");
```
Comment (rashikaranpuria) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: Entity to message conversion can be abstracted out to another method. Will improve readability of method

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (KhawYewOnn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:23]: ok

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (KhawYewOnn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: resolved

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (KhawYewOnn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: fixed

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (KhawYewOnn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:51]: Alright, added a function to handle this

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (KhawYewOnn) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: ok

**Pull request merged**